me.orszulak.keyring.json

```diff
@@ -0,0 +1,69 @@
+{
+	"app-id": "me.orszulak.keyring",
+	"runtime": "org.gnome.Platform",
+	"runtime-version": "41",
+	"sdk": "org.gnome.Sdk",
+	"command": "keyring",
+	"separate-locales": false,
+	"finish-args": [
+		"--share=ipc",
+		"--socket=x11",
+		"--socket=fallback-x11",
+		"--socket=wayland",
+		"--filesystem=xdg-download",
+		"--own-name=me.orszulak.keyring"
+	],
+	"modules": [
+		{
+			"name": "zenity",
+			"buildsystem": "meson",
+			"sources": [
+				{
+					"type": "archive",
+					"url": "https://gitlab.gnome.org/GNOME/zenity/-/archive/3.41.0/zenity-3.41.0.tar.gz",
+					"sha256": "803b1d6849f0ef0e8cca1bbe4f51f1ead7b8000540f2c593f8026f0cdf47ce74"
+				}
+			]
+		},
+		{
+			"name": "libjsoncpp",
+			"buildsystem": "cmake",
+			"config-opts": [
+				"-DCMAKE_BUILD_TYPE=release",
+				"-DBUILD_SHARED_LIBS=ON"
+			],
+			"sources": [
+				{
+					"type": "archive",
+					"url": "https://github.com/open-source-parsers/jsoncpp/archive/refs/tags/1.7.5.tar.gz",
+					"sha256": "4338c6cab8af8dee6cdfd54e6218bd0533785f552c6162bb083f8dd28bf8fbbe"
+				}
+			]
+		},
+		{
+			"name": "keyring",
+			"buildsystem": "simple",
+			"build-commands": [
+				"cp -r ./* /app/",
+				"ln -s /app/keyring /app/bin/keyring",
+				"install -Dm644 launcher_linux.png /app/share/icons/hicolor/scalable/apps/me.orszulak.keyring.png",
+				"install -Dm644 me.orszulak.keyring.desktop /app/share/applications/me.orszulak.keyring.desktop"
+			],
+			"sources": [
+				{
+					"type": "archive",
+					"url": "https://gitlab.com/api/v4/projects/30619487/packages/generic/keyring/1.2.0-158/keyring-linux-x86.tar.gz",
+					"sha256": "fae6821c91415c814c62e871cd31f5bfa53852d167dccfa458dcdfd7b70e042f"
+				},
+				{
+					"type": "file",
+					"path": "launcher_linux.png"
```

Comment:
upstream doesn't ship an icon?

Reply:
I am the upstream 😅 but unfortunately I'm not the best at designing icons 😆
If this is not the best place to keep the icon I can move that.
This is heavily inspired by how this [app](https://github.com/flathub/im.fluffychat.Fluffychat) is built because it's also in flutter

Reply:
*sigh*

Reply:
the thing is that it should be shipped upstream as well as .desktop and the .appdata.xml